### DISCUSSION TITLE
How do i use style on a scale?

### DISCUSSION THREAD

**Discussion by mttomaz** [elkowar/eww#560] [Q&A]
Sorry if it is simple, but i can't figure it out how to change the colors of a scale widget conditionally. I got a widget that i use on a sidebar for volume control, and i was trying to make it change the color of the scale if the master volume is muted. This is what i did so far: 
<img src="https://i.imgur.com/0iBcM4K.png"/>
Did someone do something similar?

**Comment by viandoxdev** [comment]
Set a class conditionally and style that class, the css that goes into `:style` isn't scss, since it's passed straight to GTK.

**Reply by mttomaz** [reply]
Ok, that's solves the problem with my widget, but i have another one that is a music widget, and i was wondering if i could get the colors from the cover and use it to theme the scales, is it possible, or it will just have to use a "static color"?

**Reply by viandoxdev** [reply]
you can probably do that with a literal widget and some imagemagick/libvip magic but i don't know how it would perform (might be a bit slow so you'd want to cache things, i.e hash name and artist and use that has a filename in a directory).

**Reply by mttomaz** [reply]
Sorry, I didn't spoke it right, i already have a script to fetch the colors. I didn't thought of using a literal widget, but then how yould the colors change dinamically? is there some way to do a temporary .scss file?

**Reply by viandoxdev** [reply]
Thats a pretty hard problem, probably something you'd be better off writing a patch for (native performance), you might be able to use gtk css variables but I doubt it. Other than that you'd have to hack something with an invisible scale over (overlay widget) a styled box that looks like the scale.
